packages/console/src/pages/SignInExperience/index.tsx

```diff
@@ -41,11 +41,19 @@ const SignInExperience = () => {
     handleSubmit,
     getValues,
     watch,
-    formState: { isSubmitting },
+    formState: { isSubmitting, isDirty },
   } = methods;
   const api = useApi();
   const formData = watch();
 
+  const previewConfigs = useMemo(() => {
+    if (!isDirty) {
+      return data;
+    }
+
+    return signInExperienceParser.toRemoteModel(formData);
+  }, [formData, isDirty, data]);
+
```

Comment:
Maybe we should extract it into a hook and re-use it in both GuideModal and SignInExperience page

Reply:
Hmm, maybe even re-use the form? It seems everything has two copies

Reply:
They two have much difference actually.

I refactored this hook logic, PTAL